### PR TITLE
[qspi][spiflash][stm32f746] Support linear mode for STM32f7 qspiflash

### PR DESCRIPTION
**Pull request by gkalsi**

### CODE REVIEW

File: platform/stm32f7xx/qspi.c [hunk 1]
```diff
@@ -375,7 +377,16 @@ static ssize_t spiflash_bdev_erase(struct bdev* device, off_t offset,
 
 static int spiflash_ioctl(struct bdev* device, int request, void* argp)
 {
-    return ERR_NOT_IMPLEMENTED;
+    int ret = ERR_NOT_SUPPORTED;
+
+    switch (request) {
+        case BIO_IOCTL_GET_MEM_MAP:
```
Comment (travisg) [inline platform/stm32f7xx/qspi.c]: There's a PUT_MEM_MAP too, should support that for when writes are needed. Unless it supports writing while in linear mode?

Comment (gkalsi) [inline platform/stm32f7xx/qspi.c]: Doesn't support writing while in linear mode.

Will add that in the next commit.

**Comment by travisg** [comment]
okay.

**Comment by travisg** [comment]
LGTM

**Comment by gkalsi** [comment]
Merged.

**Pull request closed without merging**